### PR TITLE
CFn refactoring Pt.4

### PR DESCRIPTION
**Pull request by dominikschubert**
## Changes

- Remove `PLACEHOLDER_RESOURCE_NAME` and `GenericBaseModel.get_resource_name`
- Use full resource type names everywhere (e.g. `AWS::SQS::Queue` instead of `SQS::Queue`)
- Use `GenericBaseModel.logical_resource_id` instead of `GenericBaseModel.resource_id`
- Move non-engine utils to `api_utils.py` (like in lambda)
- Move policy loading to `policy_loader.py`
- Remove `nested_stacks` as mentioned in a previous PR

### CODE REVIEW

**Review by pinzon — APPROVED** [review]
Awesome!!! :rocket:

File: localstack/services/cloudformation/engine/template_deployer.py [hunk 1]
```diff
@@ -471,8 +420,9 @@ def resolve_refs_recursively(stack, value):
 
 
 @prevent_stack_overflow(match_parameters=True)
-# TODO: move Stack model into separate file and add type hints here
-def _resolve_refs_recursively(stack, value):
+def _resolve_refs_recursively(
+    stack: Union[Stack, "TemplateDeployer"], value: dict | list | str | bytes | None
```
Comment (whummer) [inline localstack/services/cloudformation/engine/template_deployer.py:424]: nit: we could define a type alias for this type union, something like:
```
AnyTemplateValue = dict | list | str | bytes | None
```

... and then also use it as return type of this function. Not critical at all, though

**Review by whummer — APPROVED** [review]
Awesome set of cleanup changes! 🧹 Great that we are able to get rid of `PLACEHOLDER_RESOURCE_NAME`, as well as the remaining `moto` exceptions, etc. 🚀

File: localstack/services/cloudformation/engine/template_deployer.py (hunk 1 above)
Comment (dominikschubert) [reply localstack/services/cloudformation/engine/template_deployer.py:424]: :+1:  We can iterate over the typing soon. Not sure how stable this is right now so I'd leave it as it is for now

**Pull request merged**